src/preferences/preference-service.ts

```diff
@@ -0,0 +1,62 @@
+import { Emitter, Event } from "../application/common";
+
+
+export class PreferenceService {
+    protected readonly onPreferenceChangedEmitter = new Emitter<PreferenceChangeEvent>();
+
+    protected readonly preferences = new Map<string, Preference>();
+
+    protected firePreferenceChanged(pref: PreferenceChangeEvent) {
+        this.onPreferenceChangedEmitter.fire(pref);
+    }
+
+    get onPreferenceChanged(): Event<any> {
+        return this.onPreferenceChangedEmitter.event;
+    }
+
+    get preferenceNames(): IterableIterator<string> {
+        return this.preferences.keys();
+    }
+
+    /**
+     * Get preference by key if it exists, return undefined if not
+     * @param key Unique preference string identifier
+     */
+    getPref(key: string): Preference | undefined {
```

Comment:
I don't think we should expose setters for now, at least not from the same interface.
Currently I assume all changes going through the file system